src/mqtt.cpp

```diff
@@ -172,22 +172,15 @@ MqttClient::MqttClient(struct json_object *option) : _enabled(false) {
 			res = mosquitto_connect(_mcs, _host.c_str(), _port, _keepalive);
 			if (res != MOSQ_ERR_SUCCESS) {
 				switch (res) {
-				case MOSQ_ERR_CONN_REFUSED: // mqtt might accept us later only.
+				case MOSQ_ERR_CONN_REFUSED:
+				case MOSQ_ERR_ERRNO:
+					// We assume all system errors to be externally recoverable,
+					// so we retry.
 					print(log_warning, "mosquitto_connect failed (but trying anyhow): %s", "mqtt",
 						  mosquitto_strerror(res));
 					break;
-				case MOSQ_ERR_ERRNO:
-					if (errno == 111) // con refused:
```

Comment:
you're removing a bit too much code for my taste.
rather than completely removing the  handling of `MOSQ_ERR_ERRNO`, i would prefer to extend it to log the errno value (ideally with `strerror()`) (granted, we didn't do that before.).

Reply:
Luckily [mosquitto_strerror](https://github.com/eclipse/mosquitto/blob/15292b20b0894ec7c5c3d47e4b22ee9d89f91132/lib/strings_mosq.c#L31) already does that.

Reply:
Done.